test/cppsuite/README.md

```diff
@@ -1 +1,144 @@
+| Finish   | N/A |
+
+## Timestamp Manager
+The timestamp manager is responsible for managing the stable and oldest timestamps as well as providing timestamps to the user on demand. The stable and oldest timestamps are updated according to the definitions of the `oldest_lag` and `stable_lag` in the test configuration file. The `oldest_lag` is the difference in time between the oldest and stable timestamps, and `stable_lag` is the difference in time between the stable timestamp and the current time. When the timestamp manager updates the different timestamps it is possible that concurrent operations may fail. For instance, if the oldest timestamp moves to a time more recent than a commit timestamp of a concurrent transaction, the transaction will fail. 
+
+| Phase    | Description |
+| -------- | ----------- |
+| Load     | Initializes the oldest and stable lags using the values retrieved from the configuration file. |
+| Run      | Updates the oldest and stable lags as well as the oldest and stable timestamps. |
+| Finish   | N/A |
+
+## Metrics Monitor
+The metrics monitor polls database metrics during and after test execution. If a metric is found to be outside the acceptable range defined in the configuration file then the test will fail. When a test ends, the metrics monitor can export the value of these statistics to a JSON file that can be uploaded to Evergreen or Atlas.
+
+| Phase    | Description |
+| -------- | ----------- |
+| Load     | Retrieves the metrics from the configuration file that need to be monitored during and after the test. |
+| Run      | Checks runtime metrics and fails the test if one is found outside the range defined in the configuration file. |
+| Finish   | Checks post run metrics and fails the test if one is found outside the range defined in the configuration file. Exports any required metrics to a JSON file if the test is successful. |
+
+All the components and their implementation can be found in the [component](https://github.com/wiredtiger/wiredtiger/tree/develop/test/cppsuite/src/component) folder.
+
+# Test configuration file
+A test configuration is a text file that is associated with a test and defines the workload. The test configuration file contains top-level settings such as the test duration or cache size and component level settings that define their behavior. The format is the following:
+
+```cpp
+# Top level settings
+duration_seconds=15,
+cache_size_mb=200,
+...,
+# Components settings
+timestamp_manager=
+(
+  enabled=true,
+  oldest_lag=1,
+  stable_lag=1,
+  ...,
+),
+...,
+```
+
+Each test has a default configuration file which is the test name followed by the suffix `_default` and the extension `.txt`. It is possible to use any configuration file when executing a test, see [here](how_to_use_cppsuite.md#running-tests) for more details.
+
+All the different configurable items are defined in [test_data.py](https://github.com/wiredtiger/wiredtiger/blob/develop/dist/test_data.py).
```

Comment:
The configurable fields and default settings are defined in in [test_data.py](https://github.com/wiredtiger/wiredtiger/blob/develop/dist/test_data.py).

Reply:
Edited in 56d5a8190f84a920594a35e1e6c9a202fe874265